### DISCUSSION TITLE
SCA UserOperation Gas Estimation calculation

### DISCUSSION THREAD

**Discussion by AbhishekCDN** [alchemyplatform/aa-sdk#976] [Issues]
## Description

We are using the buildUserOperation function of the modular account client to fetch the Gas Estimation for a UserOperation.

We get three fields 
1. callGasLimit
2. verificationGasLimit
3. preVerificationGas

What calculations should we perform on these to get a single-value network fee estimate for UO to show on UI?

## Current Implementation

We add up all three
let network_fee = (callGasLimit + verificationGasLimit  + preVerificationGas) * gasPrice 

### Inference 
Sometimes using the above method gives a much lower value, than what we get when sending the UO.

**Comment by avarobinson** [comment]
Hi @AbhishekCDN - you should not need to do any additional estimates. 

What do you mean by "show on UI"? You can use the sendUserOperation call directly to estimate gas and send the UO. The values will fluctuate each time estimates are run depending on network activity.

**Reply by AbhishekCDN** [reply]
Hey @avarobinson ,
Well if you have used any wallet app like Trust Wallet, or Metamask
These show network fee estimates before sending any transaction right? That is what I mean for show on UI

![image](https://github.com/user-attachments/assets/ed2629a8-69c9-4153-af65-3676c21e8bf8)

like in this image the estimated fee is shown
